Stack/Opc.Ua.Core/Stack/Transport/IMessageSocket.cs

```diff
@@ -135,6 +135,15 @@ public interface IMessageSocket : IDisposable
         /// <value>The socket handle.</value>
         int Handle { get; }
 
+        /// <summary>
+        /// Gets the local endpoint.
+        /// </summary>
+        /// <exception cref="System.Net.Sockets.SocketException">An error occurred when attempting to access the socket.
+        /// See the Remarks section for more information.</exception>
+        /// <exception cref="System.ObjectDisposedException">The Socket has been closed.</exception>
+        /// <returns>The System.Net.EndPoint that the Socket is using for communications.</returns>
+        System.Net.EndPoint LocalEndpoint { get; }
```

Comment:
IMessageSocket is an abstraction for a socket, the underlying technology may not be a Net Socket. Better were to abstract the information you need from the Endpoint and return that? Or have a IMessageSocketEndpoint.

Reply:
ok, Endpoint is an abstract class, so the use should be ok here...never mind

Reply:
I also wouldn't mind including the getter only in the TcpMessageSocket and casting to a TcpMessageSocket before getting the value - I just thought since EndPoint is an abstract implementation and after all a socket should always have some sort of LocalEndPoint... but as I said I don't really mind

Reply:
Right, since its an abstract its ok...